### PR TITLE
VZ-6239.  Component level uninstall for Grafana

### PR DESCRIPTION
**Pull request by michael-cico**
Enable component-level uninstall for Grafana in the VPO.

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/common/vmi.go [hunk 1]
```diff
@@ -159,6 +177,21 @@ func EnsureGrafanaAdminSecret(cli client.Client) error {
 	return nil
 }
 
+func DeleteGrafanaAdminSecret(cli client.Client) error {
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/common/vmi.go:180]: uninstall.go also has deleteSecret, this common code should be moved to pkg/k8s/secret.go

File: platform-operator/controllers/verrazzano/component/grafana/dashboard.go [hunk 2]
```diff
@@ -5,6 +5,8 @@ package grafana
 
 import (
 	"context"
+	"k8s.io/apimachinery/pkg/api/errors"
```
Comment (davisgomes) [inline platform-operator/controllers/verrazzano/component/grafana/dashboard.go:8]: Move imports to second block

Comment (davisgomes) [reply platform-operator/controllers/verrazzano/component/grafana/dashboard.go:8]: Same comment for all other import blocks in these files

File: platform-operator/controllers/verrazzano/component/grafana/dashboard.go [hunk 3]
```diff
@@ -60,6 +62,21 @@ func createGrafanaConfigMaps(ctx spi.ComponentContext) error {
 	return err
 }
 
+func deleteGrafanaConfigMaps(ctx spi.ComponentContext) error {
```
[the remark this answers is not in the document]
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/grafana/dashboard.go:65]: Will fix in a follow-up PR

File: platform-operator/controllers/verrazzano/component/common/vmi.go (hunk 1 above)
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/common/vmi.go:179]: Will fix in a follow-up PR

**Pull request merged**